src/inventory_ui.cpp

```diff
@@ -1894,6 +1911,34 @@ bool inventory_selector::add_entry_rec( inventory_column &entry_column,
     return vis_contents;
 }
 
+bool inventory_selector::drag_drop_item( item *sourceItem, item *destItem )
+{
+    if( sourceItem == destItem ) {
+        popup( "Can't move an item onto the same item!" );
+        return false;
+    }
+    auto pockets = destItem->get_all_contained_pockets();
+    if( pockets.empty() ) {
+        popup( "Destination has no pockets." );
+        return false;
+    }
+    for( auto pocket : pockets ) {
+        ret_val<item_pocket::contain_code> contain = pocket->can_contain( *sourceItem );
+        if( contain.success() ) {
+            // storage should mimick character inserting
+            get_avatar().as_character()->store( pocket, *sourceItem );
```

Comment:
Does this handle the movement cost of moving items between pockets?

Reply:
So I copied this code from pocket_favorite_callback::move_item, the function used by the pocket organizer menu. This usage appears to hard-code a movement cost of 100. It looks like I could be calling item::insert_cost to handle this better, so if you think it's OK to change this usage in both places in this PR, I will do so.

Reply:
It would be nice to unify the cost, but before that I'm also concerned that it may not be using an activity to move the items from the sound of it, and fixing that would require more changes and is better done in a separate PR. If you move a large amount of items in these UIs and exit from them, do you have to wait for additional turns and can you interrupt the waiting?

Reply:
Unfortunately these questions are above my current level of knowledge on how the game works. Maybe a separate issue to address both use cases would be best?

Reply:
I didn't get to actually test it, yet, but from the sounds of it I don't think this should go in like this without being an actual activity. Worst case you're dragging around a bunch of items in the menu trying to figure out some ideal layout and it stacks up moves that cause you to lose turns without any clear indication why.

Either check out how wield/wear in the pickup menu does it (close menu, do activity, reopen menu) or implement this as a sort of planning tool that lists all the movement actions you want to perform with the moves it will take that requires confirmation before it's executed (as an activity so it's interruptible). Inventory management in this game is neither meant to be free nor to lock your character from reacting to things happening around you.

Reply:
What about showing a dialog upon releasing the left-click button that makes the user clicks the pocket the item would be going into that shows the movement cost? This way we would also handle things like backpacks with multiple pockets, the user could choose which pocket the item goes into while taking into consideration the cost.

Reply:
I currently have no real opinion on the ui side of the implementation. Just that the ui should communicate what's happening and that whatever is happening should be an activity.

Reply:
I'm not sure what you mean by activity. As I said above the number of moves is currently hard-coded as 100 and not shown to the user. I will display the proper value. But the way it works here is the same way it works in the Inventory Organization screen. If we need to change the work-flow of how moving items between pockets is communicated to the user, I believe it would be more appropriate to create a new issue after this PR is merged to fix it here and in the Inventory Organization screen.